Version file standardization

guidomodarelli opened the pull request:
### Description

As part of the standardization process, all XDR-SIEM repositories will use a consistent identification method. Each repository will include a standardized `VERSION.json` file containing necessary data to identify the origin branch and release stage. This file should be included in the installer's installation directory and indicate the short commit used to create it.

#### Requirements

<details><summary>GitHub</summary>

1. Each repository must include a `VERSION.json` file in the root directory.
2. The file must be named `VERSION.json`.
3. It should follow the specified `JSON` format (with required keys).
4. The `version` key value must follow the `MAJOR.MINOR.PATCH` format.
5. The `stage` key value must indicate the `release stage`, using lowercase and no spaces.
6. Numeric revisions (e.g., `41200`) should be replaced by the `stage` value from the version file.
7. References to the version and stage (revision) should be fetched from this `JSON` file.

> [NOTE]
> References such as Jenkinsfile shared libraries or `uses: repo/workflow@main` are exempted from this change (7).
> However, procedures like package generation should be adapted accordingly.
  
</details>


<details><summary>Installers</summary>

1. Installers should install the `VERSION.json` file in the installation directory:
    1. Wazuh manager and Wazuh agent: `/var/ossec` (or equivalent)
    1. Wazuh dashboard and Wazuh indexer: `/usr/share/wazuh-{dashboard/indexer}`
2. The `version` and `stage` (revision) should be fetched from the `VERSION.json` file.
3. The `VERSION.json` provided in the installer package should include the `short commit` used to create the it. If the repository requires a compound commit due to dependencies, use it instead of a short commit.
4. Installer nomenclature and metadata must remain unchanged.
5. File Permissions and Ownership for VERSION.json:
    1. Ownership: Subsystem user and group
    1. Permissions: `440`

</details>

<details><summary>Documentation</summary>

1. Documentation must be modified to reflect the changes.

</details>

#### Resources

<details><summary>VERSION.json file content and structure</summary>

- GitHub repository

```
{
    "version": "4.12.0",
    "stage": "alpha0"
}
```

- Installers

```
{
    "version": "4.12.0",
    "stage": "alpha0",
    "commit": "short_commit_value"
}
```

</details>

### Issues Resolved

<!-- List any issues this PR will resolve. -->
<!-- Example: closes #1234 -->
[Version file standardization · Issue #526 · wazuh/wazuh-dashboard](https://github.com/wazuh/wazuh-dashboard/issues/526)

## Changelog
<!--
Add a short but concise sentence about the impact of this pull request. Prefix an entry with the type of change they correspond to: breaking, chore, deprecate, doc, feat, fix, infra, refactor, test.
- fix: Update the graph
- feat: Add a new feature

If this change does not need to added to the changelog, just add a single `skip` line e.g.
- skip

Descriptions following the prefixes must be 100 characters long or less
-->
- feat: migrate versioning to JSON format and update build scripts

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

Desvelao commented:
The packages generated locally need to get the combined (Wazuh dashboad + plugins short commits) commit SHA. I am not sure if this should be supported or not.

Desvelao reviewed (approved):
LGTM!

guidomodarelli commented:
This pull request includes several changes to the build and packaging process for the Wazuh dashboard. The primary focus is on integrating version information from a new `VERSION.json` file and updating the build scripts accordingly. Additionally, there are updates to the `.eslintignore` and `.lycheeignore` files, and various improvements to the documentation.

### Build and Packaging Updates:
* Added `VERSION.json` file to store version information and updated scripts to use this file for versioning. (`VERSION.json`: [[1]](diffhunk://#diff-738b01019964420da05e7eafdf6104aa10601949594548f6785cd6559101d56eR1-R4) `.github/workflows/build_wazuh_dashboard_with_plugins.yml`: [[2]](diffhunk://#diff-28d61cadd84c3d16854ea4e62e8326245c516b09cfd406de2763fba4cbb35e05L156-R159) `dev-tools/build-packages/build-packages.sh`: [[3]](diffhunk://#diff-f6b1486cb5612e07b8561970ee326f7864cee3ebbcc476f3c4f38d0aa9a5092aL17-R21)
* Updated the `build-packages.sh` script to include a new `--commit-sha` option and removed the `--version` option. (`dev-tools/build-packages/build-packages.sh`: [[1]](diffhunk://#diff-f6b1486cb5612e07b8561970ee326f7864cee3ebbcc476f3c4f38d0aa9a5092aR178-R183) [[2]](diffhunk://#diff-f6b1486cb5612e07b8561970ee326f7864cee3ebbcc476f3c4f38d0aa9a5092aL235-L242)
* Modified the Docker build scripts to copy `VERSION.json` and include version information in the final package. (`dev-tools/build-packages/base/Docker/Dockerfile`: [[1]](diffhunk://#diff-2f719909922f6f26115bffb6075a0b4fb952bb9c7626c455cec01d751cc3acb3R31) `dev-tools/build-packages/deb/deb-builder.sh`: [[2]](diffhunk://#diff-62045b3b9ac4710ca7f1312d38d05e0afbc8fa27ec0babb9627a68d9365e5292L59-R60) `dev-tools/build-packages/rpm/rpm-builder.sh`: [[3]](diffhunk://#diff-72b468fdf38b5b389ce8a2788d3987b17002f68e55f6f4269c2a9647514726bcL59-R60)

### Workflow Enhancements:
* Added new SHA variables (`PLUGINS_SHA`, `SECURITY_SHA`, `REPORTING_SHA`) to the GitHub Actions workflow for building the Wazuh dashboard with plugins. (`.github/workflows/build_wazuh_dashboard_with_plugins.yml`: [[1]](diffhunk://#diff-28d61cadd84c3d16854ea4e62e8326245c516b09cfd406de2763fba4cbb35e05R114-R116) [[2]](diffhunk://#diff-28d61cadd84c3d16854ea4e62e8326245c516b09cfd406de2763fba4cbb35e05R196-R198)
* Adjusted the package naming logic in the workflow to include the new SHA variables. (`.github/workflows/build_wazuh_dashboard_with_plugins.yml`: [[1]](diffhunk://#diff-28d61cadd84c3d16854ea4e62e8326245c516b09cfd406de2763fba4cbb35e05L172-R185) [[2]](diffhunk://#diff-28d61cadd84c3d16854ea4e62e8326245c516b09cfd406de2763fba4cbb35e05L313-R330)

### Documentation Updates:
* Updated the `README.md` for the build packages to reflect the new `--commit-sha` option and removed the `--version` option. (`dev-tools/build-packages/README.md`: [[1]](diffhunk://#diff-20e71a17388ecd45d0e25287431d0de7b496cf69ac232609fb265d23462e0644R29-L33) [[2]](diffhunk://#diff-20e71a17388ecd45d0e25287431d0de7b496cf69ac232609fb265d23462e0644L57-R59)

### Ignore Files:
* Added `VERSION.json` to `.eslintignore` and `.gitignore` files to prevent it from being linted or accidentally committed. (`.eslintignore`: [[1]](diffhunk://#diff-a0fdacd2ade87c5238dde44378f574f7e123e669acdf8b740878b14b33b20275R37-R38) `dev-tools/build-packages/base/Docker/.gitignore`: [[2]](diffhunk://#diff-ed0d76d1cbfd905e9df64f8779d49ed429e886777b862c1417407121010a564dR1-R4)

yenienserrano reviewed (approved):
LGTM

<img width="440" alt="image" src="https://github.com/user-attachments/assets/9dcfcaaa-b6d8-498d-80c6-3cc99f2ee243" />
